### PR TITLE
Bump Android Dependency Versions

### PR DESCRIPTION
**Pull request by jamesonwilliams**
Amplify Android 1.17.0
Android Gradle Plugin 4.1.2
Core Desugaring Library 1.1.5

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by richardmcclellan — APPROVED** [review]

File: docs/lib/datastore/fragments/android/getting-started/30_platformIntegration.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.1.1'
+        classpath 'com.android.tools.build:gradle:4.1.2'
```
Comment (rjuliano) [inline docs/lib/datastore/fragments/android/getting-started/30_platformIntegration.md:16]: Is this the guy the breaks the build or did they finally patch it?

Comment (rjuliano) [reply docs/lib/datastore/fragments/android/getting-started/30_platformIntegration.md:16]: Not sure if the issue we observed when attempting to move to 4.1.2 for the library itself also happens when users are creating an app.

Comment (jamesonwilliams) [reply docs/lib/datastore/fragments/android/getting-started/30_platformIntegration.md:16]: This breaks the build _of our library itself_, but it should work fine for a consumer of our library. (I've been using it myself, in a sample app.)

Comment (richardmcclellan) [reply docs/lib/datastore/fragments/android/getting-started/30_platformIntegration.md:16]: The issue we are running into with the library is trying to jump from 4.0.x to 4.1.x, not 4.1.1 to 4.1.2, so I think this change is safe.

**Pull request merged**